src/features/boundary_selector/atoms/boundaryMarkerAtom.ts

```diff
@@ -142,6 +142,11 @@ export const boundaryMarkerAtom = createAtom(
                 [
                   // Reset button state to default
                   boundarySelectorControl.setState('regular'),
+                  // Clear highlightedGeometry
+                  highlightedGeometryAtom.set({
```

Comment:
We can use util class for that:
highlightedGeometryAtom.set(new FeatureCollection([]))

Reply:
thank you, fixed